x-pack/plugins/ml/public/application/components/data_grid/column_chart.tsx

```diff
@@ -0,0 +1,126 @@
+  isOrdinalChartData,
+  useColumnChart,
+  ChartData,
+} from './use_column_chart';
+
+const getLegendText = (chartData: ChartData, MAX_CHART_COLUMNS: number): string | JSX.Element => {
+  if (chartData.type === 'boolean') {
+    return (
+      <table className="mlDataGridChart__legendBoolean">
+        <tbody>
+          <tr>
+            <td>{chartData.data[0].key_as_string}</td>
+            <td>{chartData.data[1].key_as_string}</td>
+          </tr>
+        </tbody>
+      </table>
+    );
+  }
+
+  if (isOrdinalChartData(chartData) && chartData.cardinality === 1) {
+    return i18n.translate('xpack.ml.dataGridChart.singleCategoryLegend', {
+      defaultMessage: `{cardinality} category`,
+      values: { cardinality: chartData.cardinality },
+    });
+  }
+
+  if (isOrdinalChartData(chartData) && chartData.cardinality > MAX_CHART_COLUMNS) {
+    return i18n.translate('xpack.ml.dataGridChart.topCategoriesLegend', {
+      defaultMessage: `top {MAX_CHART_COLUMNS} of {cardinality} categories`,
+      values: { cardinality: chartData.cardinality, MAX_CHART_COLUMNS },
+    });
+  }
+
+  if (
+    isOrdinalChartData(chartData) &&
+    chartData.cardinality <= MAX_CHART_COLUMNS &&
+    chartData.cardinality > 1
+  ) {
+    return i18n.translate('xpack.ml.dataGridChart.categoriesLegend', {
+      defaultMessage: `{cardinality} categories`,
+      values: { cardinality: chartData.cardinality },
+    });
+  }
+
+  if (isNumericChartData(chartData)) {
+    return `${Math.round(chartData.stats[0] * 100) / 100} - ${
+      Math.round(chartData.stats[1] * 100) / 100
+    }`;
+  }
+
+  throw new Error('Invalid chart data.');
+};
+
+interface Props {
+  chartData: ChartData;
+  columnType: EuiDataGridColumn;
+}
+
+export const ColumnChart: FC<Props> = ({ chartData, columnType }) => {
+  const { coloredData, xScaleType, MAX_CHART_COLUMNS } = useColumnChart(chartData, columnType);
+
+  if (coloredData.length === 0) {
+    return columnType.id;
+  }
+
+  return (
+    <>
+      <div className="mlDataGridChart__histogram">
+        <Chart>
+          <Settings
+            theme={{
+              chartMargins: {
+                left: 4,
+                right: 4,
+                top: 5,
+                bottom: 1,
+              },
+              chartPaddings: {
+                left: 0,
+                right: 0,
+                top: 0,
+                bottom: 0,
+              },
+              scales: { barsPadding: 0.1 },
+            }}
+          />
+          <BarSeries
+            id="histogram"
+            name="count"
+            xScaleType={xScaleType}
+            yScaleType="linear"
+            xAccessor="key"
+            yAccessors={['doc_count']}
+            styleAccessor={(d) => d.datum.color}
+            data={coloredData}
+          />
+        </Chart>
+      </div>
+      <div
+        className={`mlDataGridChart__legend${
+          isNumericChartData(chartData) ? ' mlDataGridChart__legend--numeric' : ''
+        }`}
```

Comment:
would be better to use `classnames` package

Reply:
Fixed in 17b8752.